siege_utilities/geo/spatial_data.py

```
@@ -425,6 +523,17 @@ def construct_download_url(self, year: int, boundary_type: str, state_fips: Opti
                 state_info = get_fips_info(state_fips)
                 log.info(f"Constructing URL for {state_info['name']} ({state_info['abbreviation']})")
 
+            # VTD data lives in the decennial PL 94-171 redistricting product,
+            # NOT in the standard annual TIGER release.  Route before doing
+            # directory discovery so we never try to find VTD20/ on tiger.
+            if self._is_vtd_pl_boundary(boundary_type, year):
+                if not state_fips:
+                    raise BoundaryInputError(
+                        "VTD boundaries require a state_fips code.",
+                        context=ctx,
+                    )
```

Ruff (0.15.10)

[warning] Abstract `raise` to an inner function

(TRY301)